---
layout: default
---

## Resources for the workshop

  * Slides

### Sequence analysis

  * [FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format)
  * [FASTA format](https://en.wikipedia.org/wiki/FASTA_format)
  * [FASTQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) - Quality control for FASTQ data

### Sequence alignment

  * [File format specifications for SAM and VCF](http://samtools.github.io/hts-specs/)
  * [BWA (Burrows-Wheeler Aligner)](http://bio-bwa.sourceforge.net/)
  * [samtools](http://www.htslib.org/)

### Variant calling

  * [samtools](http://www.htslib.org/workflow/#mapping_to_variant)
  * [freebayes](https://github.com/ekg/freebayes)

## Continuing education

### Sequence alignment

  * [BLASR](https://github.com/PacificBiosciences/blasr) - Long-read aligner originally designed for PacBio data
  * [STAR](https://github.com/alexdobin/STAR) - RNA-seq aligner
  * [TopHat](https://ccb.jhu.edu/software/tophat/index.shtml) - RNA-seq aligner

### Variant calling

  * [GATK](https://www.broadinstitute.org/gatk/)
  * [WHAM](http://zeeev.github.io/wham/)
  * [Delly](https://github.com/tobiasrausch/delly)
  * [LUMPY](https://github.com/arq5x/lumpy-sv)
  * [VariationHunter](http://variationhunter.sourceforge.net/Home)

### De novo assembly

  * Celera / Canu
  * FALCON
  * minimus2
  * SOAPdenovo
  * AbySS
