<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Introduction to Bioinformatics</title>

		<meta name="description" content="A framework for easily creating beautiful presentations using HTML">
		<meta name="author" content="Hakim El Hattab">

		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/black.css" id="theme">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">
				<section>
					<h1>Introduction to Bioinformatics</h1>
                    <p>April 7, 2016</p>
					<p>Facilitated by <a href="http://github.com/huddlej">John Huddleston</a><br />huddlej@gmail.com</p>
				</section>

                <section>
                    <h2>Why are we here?</h2>
                    <ul>
                        <li>Experience practical bioinformatics</li>
                        <li>Explore bioinformatics tools to apply to future projects</li>
                    </ul>
                </section>

                <section>
                    <h2>Who am I?</h2>
                    <ul>
                        <li>Bioinformatics specialist in a human genetics lab at the University of Washington for 5 years</li>
                        <li>Professional programmer for 8 years</li>
                        <li>M.S. in computer science and biology from WWU</li>
                        <li>Husband, father, and nerd.</li>
                    </ul>
                </section>

                <section>
                    <h2>Who are you?</h2>
                    <ul>
                        <li>Biologists?</li>
                        <li>Computer scientists?</li>
                        <li>Students, professors, or staff?</li>
                        <li>Eager future bioinformaticians?</li>
                    </ul>
                </section>

				<section>
					<h2>The plan for today</h2>
                    <ol>
                        <li>Introductions</li>
                        <li>Working with sequence data</li>
                        <li>Creating and assessing sequence alignments</li>
                        <li>Discovering and assessing genetic variants</li>
                        <li>Review</li>
                    </ol>
				</section>

                <section>
                    <h2>Outcomes for this workshop</h2>

                    <p>At the end of this workshop you will know how to:</p>
                    <ul>
                        <li>Identify genomic variants from sequence data</li>
                        <li>Interpret and evaluate the quality of genomic data</li>
                        <li>Apply tools that rely on standard genomic formats</li>
                    </ul>
                </section>

				<section>
					<h2>Review of command line basics</h2>

                    <p>Let's review before we begin...</p>
				</section>

				<section>
					<h2>Login to Amazon workstation</h2>

                    <p>Open a terminal and run the following command:</p>
                    <p><small><code>ssh -i key-pair.pem participantN@ip.address</code></small></p>
                    <p>Change to your working directory:</p>
                    <p><small><code>cd analysis/</code></small></p>
				</section>

				<section>
					<h2>Why do we sequence organisms?</h2>
                    <ul>
                        <li>What questions do we have about our organisms?</li>
                        <li>How can sequence data answer those questions?</li>
                        <li>What questions can't be answered by sequence data?</li>
                    </ul>
                    <p class="fragment"><img src="images/sequence_all_the_things.jpg" width="300" height="300" /></p>
				</section>

				<section>
					<h2>How do sequences originate?</h2>
                    <p>
                      <img src="images/Kircher_et_al_2011_Illumina_sequencing.jpg" width="60%" height="60%" /><br />
                      <small>Kircher et al. 2011</small>
                    </p>
                    <p><small>(<a target="_new" href="http://www.illumina.com/content/dam/illumina-marketing/documents/applications/ngs-library-prep/ForAllYouSeqMethods.pdf">Bonus poster</a> if you really want your minds blown)</small></p>
				</section>

				<section>
					<h2>What do we do with sequences once we have them?</h2>
                    <ul>
                        <li class="fragment">Assess for quality</li>
                        <li class="fragment">Align to a reference assembly and call variants</li>
                        <li class="fragment">Align to databases of previously annotated sequences (e.g., BLAST)</li>
                        <li class="fragment">Assemble into a representation of the original DNA</li>
                        <li class="fragment">Annotate genomic or functional features</li>
                        <li class="fragment">Search for specific sequences (e.g., motifs, kmers, etc.)</li>
                        <li class="fragment">Infer population structure</li>
                        <li class="fragment">And...?</li>
                    </ul>
				</section>

				<section>
					<h2>What we will do today</h2>
                    <ul>
                        <li>Assess for quality</li>
                        <li>Align to a reference assembly and call variants</li>
                    </ul>
				</section>

				<section>
					<h2>Sequence review</h2>
                    <ul>
                        <li>What do "sequence" data look like?</li>
                        <li class="fragment">How big are our FASTQ files?</li>
                        <li class="fragment">How do we evaluate sequence quality?</li>
                    </ul>

                    <p class="fragment"><code>fastqc --help</code></p>
				</section>

				<section>
					<h2>Sequence alignment</h2>
                    <ul>
                        <li>Why would we align our sequences to a "reference" genome?</li>
                        <li class="fragment">What does a reference look like?</li>
                        <li class="fragment">How do we align sequences to a reference?</li>
                    </ul>

                    <p class="fragment">
                        <code>samtools faidx</code><br />
                        <span class="fragment"><code>bwa</code><br /></span>
                        <span class="fragment"><code>bwa index</code><br /></span>
                        <span class="fragment"><code>bwa mem</code></span>
                    </p>
				</section>

				<section>
					<h2>Inspecting sequence alignments</h2>
                    <ul>
                        <li>How long did the alignment take?</li>
                        <li>How big is the output file?</li>
                        <li>What type of file is the output?</li>
                        <li>How many lines are in the file?</li>
                        <li>How does the file appear to be organized?</li>
                    </ul>
				</section>

				<section>
					<h2>Working with SAM files</h2>

                    <ul>
                        <li>First, let's review the SAM specification</li>
                        <li class="fragment">How do you view just the SAM header?</li>
                        <li class="fragment">How many reads are in your SAM file?</li>
                        <li class="fragment">How many unmapped reads are in the SAM file?</li>
                    </ul>

                    <p class="fragment">
                        <code>samtools</code><br />
                        <span class="fragment"><code>samtools view</code><br /></span>
                        <span class="fragment"><code>samtools flags</code></span>
                    </p>
				</section>

				<section>
					<h2>Converting from SAM to BAM</h2>

                    <p>Most tools require sorted, indexed BAMs</p>
                    <p class="fragment">
                        <code>samtools sort</code><br />
                        <span class="fragment"><code>samtools index</code><br /></span>
                    </p>

                    <ul class="fragment">
                        <li>How big is the BAM file and its index?</li>
                        <li>How can you view the alignments in the BAM file?</li>
                    </ul>
				</section>

				<section>
					<h2>Summarizing alignments</h2>

                    <p>How do we assess the quality of alignments?</p>
                    <p class="fragment">
                        <code>samtools idxstats</code><br />
                        <span class="fragment"><code>samtools flagstat</code><br /></span>
                        <span class="fragment"><code>samtools stats</code><br /></span>
                        <span class="fragment"><code>plot-bamstats</code><br /></span>
                    </p>
				</section>

				<section>
					<h2>Visualizing alignments</h2>

                    <p>What do alignments look like?</p>
                    <p class="fragment"><code>samtools tview</code></p>
                    <p class="fragment">What can we learn from visualizing alignments like this?</p>
				</section>

				<section>
					<h2>Calling variants</h2>

                    <ul>
                        <li>How would you design a tool to call SNVs and indels?</li>
                        <li class="fragment">What are the constraints on size and content of SNVs?</li>
                        <li class="fragment">What are the constraints on the size of indels?</li>
                    </ul>

                    <p class="fragment">
                        <code>freebayes</code><br />
                        <span class="fragment"><code>bgzip</code><br /></span>
                        <span class="fragment"><code>tabix</code><br /></span>
                    </p>
				</section>

				<section>
					<h2>Inspecting variants</h2>
                    <ul>
                        <li>How long did variant calling take?</li>
                        <li>How big is the output file?</li>
                        <li>What type of file is the output?</li>
                        <li>How many lines are in the file?</li>
                        <li>How does the file appear to be organized?</li>
                    </ul>
				</section>

				<section>
					<h2>Interpreting VCF files</h2>

                    <ul>
                        <li>First, let's review the VCF specification</li>
                        <li class="fragment">What biological patterns do you observe in the VCF?</li>
                        <li class="fragment">Can you find the two artificial mutations introduced into the reference?</li>
                        <li class="fragment">How do you interpret the SNVs and indels we didn't introduce?</li>
                        <li class="fragment">How do these results affect your interpretation of initial variant caller output?</li>
                        <li class="fragment">How could we establish a baseline "truth" for variant calls?</li>
                    </ul>
				</section>

				<section>
					<h2>Summarizing variants</h2>

                    <p>How do we assess the quality of variants?</p>
                    <p class="fragment">
                        <code>bcftools stats</code><br />
                        <code>vcfstats</code>
                    </p>
				</section>

				<section>
					<h2>Working with VCF files</h2>

                    <ul>
                        <li>Simplify the INFO field to just the allele count (AC), read depth (DP), and variant type (TYPE) fields.</li>
                        <li class="fragment">Find all indels with quality >200.</li>
                        <li class="fragment">Find all variants with a homozygous alternate genotype (i.e., "1/1").</li>
                        <li class="fragment">Combine the previous three filters in one command and output a table.</li>
                        <li class="fragment">What patterns do you notice in the final filtered results?</li>
                    </ul>

                    <p class="fragment">
                        <code>vcfkeepinfo</code><br />
                        <span class="fragment"><code>bcftools filter</code><br /></span>
                        <span class="fragment"><code>vcf2tsv</code><br /></span>
                    </p>
				</section>

				<section>
					<h2>Exploring more VCF tools</h2>

                    <ul>
                        <li>bcftools</li>
                        <li>vcflib</li>
                        <li>SnpSift</li>
                        <li>Many more...</li>
                    </ul>
				</section>

				<section>
					<h2>Let's review</h2>

                    <p>After three hours, you now know how to:</p>
                    <ul>
                        <li>Identify genomic variants from sequence data</li>
                        <li class="fragment">Interpret and evaluate the quality of genomic data</li>
                        <li class="fragment">Apply tools that rely on standard genomic formats</li>
                    </ul>

                    <p class="fragment">But this is just the beginning...</p>
				</section>

				<section>
					<h2>Plan to expand your skills</h2>

                    <ul>
                        <li>Skim the tools listed under the website's "Continuing education" section.</li>
                        <li>Take 5 minutes to brainstorm how you can apply these tools to your own research questions.</li>
                        <li>Reflect on who you know with expertise in each subject area.</li>
                    </ul>
				</section>

				<section>
					<h2>Wrapping up</h2>

                    <ul>
                        <li>Questions?</li>
                    </ul>
				</section>
			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				transition: 'slide', // none/fade/slide/convex/concave/zoom

				// Optional reveal.js plugins
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true },
					{ src: 'plugin/notes/notes.js', async: true }
				]
			});

		</script>

	</body>
</html>
